crates/typst-library/src/layout/page.rs

```diff
@@ -684,6 +713,99 @@ impl Parity {
     }
 }
 
+/// Specification for logical page numbers (PDF only).
+#[derive(Debug, Clone, PartialEq, Hash, Default)]
+pub struct LogicalNumbering {
+    /// Can be any string or none. Will always be prepended to the numbering style.
+    prefix: Option<String>,
+
+    /// Based on the numbering pattern.
+    ///
+    /// If `none` or numbering is a function, field will be empty.
+    style: Option<LabelStyle>,
+
+    /// Offset for the page label start.
+    ///
+    /// Describes where to start counting from when setting a style.
+    /// (Has to be greater or equal than 1)
+    offset: Option<NonZeroUsize>,
+}
+
+impl LogicalNumbering {
+    pub fn apply(numbering: Option<Numbering>, page: usize) -> LogicalNumbering {
```

Comment:
`apply` is a strange name, since this is a constructor basically.

Reply:
Changed it, this was a remnant when I still intended to use `impl From` before I realised I needed the page parameter.